templates/operations.ts.hbs

```diff
@@ -180,10 +179,18 @@
       const response = await fetch(req);
       if (rawResponse) {
         return response;
+      } else if (this.clientConfig.throwOnBadResponse && !response.ok && response.status !== 304) {
+        throw new Error('Error ' + response.status + ': ' + response.statusText);
       } else {
-        return response.json();
+        //If rawResponse and throwOnBadResponse is false, go with default behavior.
+        {{#if (eq (getReturnTypeFromOperation this) "void") }}
+          return;
+        {{else}}
+          const text = await response.text();
+          // It's ideal to get "{}" for an empty response body, but we won't throw if it's truly empty
+          return text ? JSON.parse(text) : {};
+        {{/if}}
```

Comment:
After the refactor, this is the last block in the function. The return statement for `void` has become redundant, because there's literally nothing after it. The whole `else` code block can be moved into the non-void return type template block.